nanoFramework.Json/TimeExtensions.cs

```diff
@@ -56,13 +56,23 @@ public static DateTime FromiCalendar(string iCalendar)
 
 			// We now have the time string to parse, and we'll adjust
 			// to UTC or timezone after parsing
-			string year = time.Substring(0, 4);
-			string month = time.Substring(4, 2);
-			string day = time.Substring(6, 2);
-			string hour = time.Substring(9, 2);
+			string year   = time.Substring(0, 4);
+			string month  = time.Substring(4, 2);
+			string day    = time.Substring(6, 2);
+			string hour   = time.Substring(9, 2);
 			string minute = time.Substring(11, 2);
 			string second = time.Substring(13, 2);
 
+			// If the input string is non-numeric, Convert.ToInt32() below will throw an exception from the Native code
+			// These exceptions cause annoying delays & output when running in the nanoFramework debugger
+			// Bail out if the input string is non-numeric
+			if (!IsNumeric(year))   return DateTime.MaxValue;
```

Comment:
we're not using 1 line if. Please make it multyline with brakets:

```csharp
if (!IsNumeric(year))
{
  DateTime.MaxValue;
}

```

and all } shloud be followed by an extra enter.

Reply:
Done